src/tests/ftest/io/dfuse_find_cmd.py

```diff
@@ -0,0 +1,299 @@
+
+        return profiler
+
+    def _setup_containers(self, dfs_path, cont_count, dfuses, containers):
+        """
+        Setup as many containers as the test requested. Return the paths where
+        the containers were mounted.
+        """
+        mount_dirs = []
+        for count in range(cont_count):
+            self.add_container(self.pool)
+            cont_dir = "{}_daos_dfuse_{}".format(self.pool.uuid, count)
+            mount_dir = os.path.join(dfs_path, cont_dir)
+            self.log.info(
+                "Creating container Pool UUID: %s Con UUID: %s",
+                self.pool, self.container)
+            self.start_dfuse(
+                self.hostlist_clients, self.pool, self.container, mount_dir)
+            dfuses.append(self.dfuse)
+            containers.append(self.container)
+            mount_dirs.append(self.dfuse.mount_dir.value)
+
+        return mount_dirs
+
+    def _create_dir_forest(
+            self,
+            paths,
+            height,
+            subdirs,
+            files_per_node,
+            needles):
+        """Create a directory tree on each path listed in the paths variable"""
+
+        remote_pythonpath = ":".join(sys.path)
+
+        count = 0
+        for path in paths:
+            self.log.info("Populating: %s", path)
+            prefix = "t{:05d}_".format(count)
+            count += 1
+            remote_args = "{} {} {} {} {} {}".format(
+                path, height, subdirs, files_per_node, needles, prefix)
+            dir_tree_cmd = "PYTHONPATH={} python {} {}".format(
+                remote_pythonpath, os.path.abspath(__file__), remote_args)
+            self._run_cmd(u"{0}".format(dir_tree_cmd))
+
+    def _run_cmd(self, cmd):
+        ret_code = general_utils.pcmd(self.hostlist_clients, cmd, timeout=180)
+        if 0 not in ret_code:
+            error_hosts = NodeSet(
+                ",".join([str(v) for k, v in ret_code.items() if k != 0]))
+            raise CommandFailure(
+                "Error running '{}' on the following hosts: {}".format(
+                    cmd, error_hosts))
+
+    def _teardown_dfuse(self, dfuses):
+        """Unmount all the containers that were created for the test"""
+        for dfuse in dfuses:
+            dfuse.stop()
+        self.dfuse = None
+
+    def _setup_challenger(self, test_path, directories):
+        """
+        Create the paths where the directory trees of the challenger will be
+        created.
+        """
+        challenger_dirs = []
+
+        self.log.info("Challenger test root %s", test_path)
+
+        for directory in range(directories):
+            dir_name = "test_dir_{:05d}".format(directory)
+            dir_name = os.path.join(test_path, dir_name)
+            self._run_cmd(u"mkdir -p {0}".format(dir_name))
+            challenger_dirs.append(dir_name)
+
+        return challenger_dirs
+
+    @classmethod
+    def _generate_temp_path_name(self, root, prefix):
```

Comment:
(pylint-bad-classmethod-argument) Class method _generate_temp_path_name should have 'cls' as first argument